cmd/soroban-cli/src/commands/network/mod.rs

```diff
@@ -97,6 +97,17 @@ impl Cmd {
                 cmd.run(global_args).await?;
             }
         };
+
+        // Additional logic for checking network requirement
+        // If WASM file is provided, don't require network arguments
+        if global_args.wasm.is_some() {
+            return Ok(());
+        }
+
+        // If no wasm is provided, network arguments are required
+        if global_args.rpc_url.is_none() || global_args.network_passphrase.is_none() || global_args.network.is_none() {
+            return Err(Error::Network);
+        }
```

Comment:
This code will run after any of the `stellar network` subcommands, not changing the behavior or arguments of `stellar contract info`. Furthermore, it uses fields that don't exist.
```
error[E0609]: no field `wasm` on type `&commands::global::Args`
   --> cmd\soroban-cli\src\commands\network\mod.rs:103:24
    |
103 |         if global_args.wasm.is_some() {
    |                        ^^^^ unknown field
    |
    = note: available fields are: `locator`, `filter_logs`, `quiet`, `verbose`, `very_verbose` ... and 2 others

error[E0609]: no field `rpc_url` on type `&commands::global::Args`
   --> cmd\soroban-cli\src\commands\network\mod.rs:108:24
    |
108 |         if global_args.rpc_url.is_none() || global_args.network_passphrase.is_none() || global_args.network.is_none() {
    |                        ^^^^^^^ unknown field
    |
    = note: available fields are: `locator`, `filter_logs`, `quiet`, `verbose`, `very_verbose` ... and 2 others

error[E0609]: no field `network_passphrase` on type `&commands::global::Args`
   --> cmd\soroban-cli\src\commands\network\mod.rs:108:57
    |
108 |         if global_args.rpc_url.is_none() || global_args.network_passphrase.is_none() || global_args.network.is_none() {
    |                                                         ^^^^^^^^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `locator`, `filter_logs`, `quiet`, `verbose`, `very_verbose` ... and 2 others

error[E0609]: no field `network` on type `&commands::global::Args`
   --> cmd\soroban-cli\src\commands\network\mod.rs:108:101
    |
108 |         if global_args.rpc_url.is_none() || global_args.network_passphrase.is_none() || global_args.network.is_none() {
    |                                                                                                     ^^^^^^^ unknown field
    |
    = note: available fields are: `locator`, `filter_logs`, `quiet`, `verbose`, `very_verbose` ... and 2 others
    ```

Reply:
Would like to add that the fixes for this issue is not very obvious, due to clap limitations. We might want to hold this one off until we do some code reorganization

Reply:
I add fixes for this issue